### ISSUE TITLE
[dualtor] ECMP nexthop hashing does not distribute the packets equally

### ISSUE THREAD

**Issue by vaibhavhd** [Azure/sonic-buildimage#7616]
#### Steps to reproduce the issue:
1. Set all MUXes to active. Start traffic from T1 to active TOR.
2. Add a route with 4 nexthops (4 different server IPs connected to downlinks).
3. Check if traffic is equally distributed among 4 downlinks.
4. Simulate one MUX (port connected to one of the nexthops) to standby.
5. Traffic will NOT be equally distributed to 3 downlinks  (for 3 active mux) and 1 uplink (for standby mux).

I tried changing ECMP hashing parameters in the packets to hash based on changing src-ip, src-port and dst-port, but the distribution remained too less on uplink ports.

#### Describe the results you received:
Packets not equally distributed among uplinks and downlinks (expected equal share across 3 downlinks each, and 1 equal share across 4 uplinks combined.
But, the distribution to uplinks are too less:
```
17:05:14 dual_tor_utils.check_nexthops_balance    L0767 INFO   | Packets received on downlink port 24: 3194
17:05:14 dual_tor_utils.check_nexthops_balance    L0767 INFO   | Packets received on downlink port 1: 3192
17:05:14 dual_tor_utils.check_nexthops_balance    L0767 INFO   | Packets received on downlink port 23: 3191
17:05:14 dual_tor_utils.check_nexthops_balance    L0777 INFO   | Packets received on uplink port 28: 12
17:05:14 dual_tor_utils.check_nexthops_balance    L0777 INFO   | Packets received on uplink port 30: 12
17:05:14 dual_tor_utils.check_nexthops_balance    L0777 INFO   | Packets received on uplink port 29: 12
17:05:14 dual_tor_utils.check_nexthops_balance    L0777 INFO   | Packets received on uplink port 31: 387
```

Route added:
```
root@str2-7050cx3-acs-02:~# ip route show 1.1.1.2                      
1.1.1.2 
  nexthop via 192.168.0.3 dev Vlan1000 weight 1 
  nexthop via 192.168.0.25 dev Vlan1000 weight 1 
  nexthop via 192.168.0.2 dev Vlan1000 weight 1 
  nexthop via 192.168.0.24 dev Vlan1000 weight 1 
root@str2-7050cx3-acs-02:~# 
```
Route object (200001):
```
root@str2-7050cx3-acs-02:~# bcmcmd  "l3 defip show" | grep "1\.1\.1\.2"
17    0        1.1.1.2/32           00:00:00:00:00:00 200001    0     0     0    0 y      (ECMP)
```

_Before MUX toggle to standby, the interface objects are all downlinks (400010 400032 400033 400033)_
```
root@str2-7050cx3-acs-02:~# bcmcmd "l3 multipath  show"
l3 multipath  show
Multipath Egress Object 200001
Interfaces: 400010 400028 400032 400033
Use count: 1
Multipath Egress Object 202048
Interfaces: 132825 132827 132829 132831
Use count: 2
drivshell>
```

_After MUX toggle to standby, the egress object is programmed correctly for 3 downlink (400010 400032 400033) and 1 uplink(202048)_
```
root@str2-7050cx3-acs-02:~# bcmcmd "l3 multipath  show"
l3 multipath  show
Multipath Egress Object 200001
Interfaces: 400010 400032 400033 202048
Use count: 1
Multipath Egress Object 202048
Interfaces: 132825 132827 132829 132831
Use count: 2
drivshell>
```

Therefore, it looks like the interfaces in multipath egress object change as expected when MUX is toggled, however it is just the distribution which is unequal.

#### Describe the results you expected:
The ECMP hashing algorithm to equally distribute the packets to interface objects based on changing 5 tuples in the traffic flow for same destination.

#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

**Comment by vaibhavhd** [comment]
I identified a test issue which do not capture all packets sent on uplinks, but even then the packet distribution to uplink ports remains unequal.

For Hierarchical ECMP, CSP is now raised with Broadcom - `CS00012189203`

**Comment by anshuv-mfst** [comment]
Issue Triage Meeting 5/26: 
 Tracked via CS00012189203

**Comment by gechiang** [comment]
This issue was fixed as part of the SAI 4.3
